frontends/p4/typeMap.cpp

```diff
@@ -50,6 +50,18 @@ bool TypeMap::isCompileTimeConstant(const IR::Expression* expression) const {
     return result;
 }
 
+// Method sets left value of expression arg if orig arg
+// is a left value.
+void TypeMap::copyIfOrigLeftValue(const IR::Expression* expression,
```

Comment:
The name of this function is not correct, because this copies the properties of the orig expression always, not only if it is a left value. It also copies the CompileTimeConstant property of the expression.

The reason this code is duplicated in the first place is that we do not have any tools to find code duplication in the compiler. And people will not use this function if they don't know that it exists, so the benefits of having a single copy are minor.

If you really want to remove the duplication you can perhaps call this function cloneExpressionProperties(to, from). But we have so many open bugs to fix that I think the benefits of such code cleanup are dwarfed by the time cost to find the duplicates and review the fixes.